src/material/list/selection-list.ts

```diff
@@ -256,6 +257,8 @@ export class MatListOption extends _MatListOptionMixinBase implements AfterConte
 
   _handleClick() {
     if (!this.disabled) {
+      if (!this.selectionList.multiple && this.selected) { return; }
```

Comment:
The early return here seems weird since there are only two statements under it. You could roll this `if` into the one above it.